php/WC_Gateway_Blockonomics.php

```diff
@@ -114,16 +114,34 @@ public function handle_requests()
         include_once 'Blockonomics.php';
         $blockonomics = new Blockonomics;
 
+        $order_hash = "";
```

Comment:
The order_hash flag is adding lots of unnecessary code. It is also not required to decrypt the order hash with select_crypto. Please refactor decrypting the hash to where it is required:
```
else if ($show_order && $crypto) {
    $order_id = $blockonomics->decrypt_hash($order_hash);
    $blockonomics->load_checkout_template($order_id, $crypto);
}else if ($select_crypto) {
    $blockonomics->load_blockonomics_template('crypto_options');
}else if ($finish_order) {
    $order_id = $blockonomics->decrypt_hash($order_hash);
    $blockonomics->redirect_finish_order($order_id);
}else if ($get_order && $crypto) {
    $order_id = $blockonomics->decrypt_hash($order_hash);
    $blockonomics->get_order_info($order_id, $crypto);
}
```

Reply:
Done